src/api/treasury/getTreasury.ts

```diff
@@ -17,128 +14,115 @@ import {
-      } else {
-        tokensByChain[chain][token.address.toLowerCase()] = {
-          ...pick(token, ['name', 'address', 'decimals']),
-          oracleId: token.oracleId ?? key,
-          oracleType: token.oracle ?? 'tokens',
-          assetType: 'token',
-        };
-      }
+function updateAssetsByChain() {
+  const tokenAssets = getTokenAddressesByChain();
+  const vaultAssets = getVaultAddressesByChain();
+
+  assetsByChain = keysToObject(ApiChains, chain => {
+    return {
+      ...(tokenAssets[chain] || {}),
+      ...(vaultAssets[chain] || {}),
+    };
+  });
+}
+
+// Load token address
+function getTokenAddressesByChain(): TreasuryAssetRegistry {
+  return keysToObject(ApiChains, chain => {
+    const tokens: Record<string, TreasuryAsset> = {};
+
+    for (const [tokenAddress, token] of Object.entries(getTokensForChain(chain))) {
+      // WNATIVE/NATIVE: duplicated as WBNB/BNB, WETH/ETH etc
+      if (['WNATIVE', 'NATIVE'].includes(token.id)) continue;
+      // WCELO/WMETIS: duplicate as same as native
+      if (
+        [
+          '0x471EcE3750Da237f93B8E339c536989b8978a438',
+          '0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000',
+        ].includes(token.address)
+      )
+        continue;
+
+      tokens[tokenAddress] = {
+        name: token.name,
+        address: token.address,
+        decimals: token.decimals,
+        assetType: isTokenNative(token) ? 'native' : 'token',
+        oracleId: token.oracleId,
+        oracleType: token.oracle,
+      };
     }
 
     if (hasChainValidator(chain)) {
-      tokensByChain[chain]['validator'] = getChainValidator(chain);
+      tokens['validator'] = getChainValidator(chain);
     }
+
+    return tokens;
   });
-  return tokensByChain;
-};
-
-// Periodically update vault deposit tokens and mooTokens
-const updateVaultTokenAddresses = () => {
-  console.log('updating treasury vault addresses');
-  const vaults = getMultichainVaults();
-  Object.keys(assetsByChain).forEach(chain => {
-    //App and API name harmony differently
-    const normalizedChainName = chain === 'one' ? 'harmony' : chain;
-
-    const chainVaults = vaults.filter(vault => vault.chain === normalizedChainName);
-    const chainTokens = assetsByChain[chain];
-
-    chainVaults.forEach(vault => {
-      // if vault receives native input, field will be missing in vault object
-      if (vault.tokenAddress) {
-        const depositTokenAddress = vault.tokenAddress.toLowerCase();
-        if (!chainTokens[depositTokenAddress]) {
-          chainTokens[depositTokenAddress] = {
-            oracleId: vault.oracleId,
-            oracleType: vault.oracle,
-            assetType: 'token',
-            decimals: vault.tokenDecimals,
-            name: vault.token,
-            address: vault.tokenAddress,
-          };
-        }
-      }
+}
 
-      const vaultAddress = vault.earnContractAddress.toLowerCase();
-      chainTokens[vaultAddress] = {
+function getVaultAddressesByChain(): TreasuryAssetRegistry {
```

Comment:
I think I'd rather have all mootokens loaded in the token service and maybe stored separately/flagged so they can be retrieved independently and "token loading" is fixed in that section of the api. wdyt?

Reply:
The vaults endpoint adds PPFS which is used by treasury